mathics/builtin/graphics.py

```diff
@@ -1268,14 +1284,24 @@ def _get_image_size(self, options, graphics_options, max_width):
                 aspect = None
 
         image_size = graphics_options['System`ImageSize']
-        image_size = image_size.get_name()
-        base_width, base_height = {
-            'System`Automatic': (400, 350),
-            'System`Tiny': (100, 100),
-            'System`Small': (200, 200),
-            'System`Medium': (400, 350),
-            'System`Large': (600, 500),
-        }.get(image_size, (None, None))
+        try:
+            if image_size.is_numeric():
+                base_width = int(image_size.to_number())
+                base_height = None  # will be computed later in calc_dimensions
+            elif image_size.get_head_name() == 'System`List':
+                base_width, base_height = ([x.to_number() for x in image_size.leaves] + [0, 0])[:2]
```

Comment:
To avoid length checking here you can use `elif image_size.has_form('List', 2)`.